### PR TITLE
feat(TUX-1175): Initial package for storybook doc components

### PR DESCRIPTION
**Pull request by Benrajalu**
**What is the problem this PR is trying to solve?**
We have many storybooks and they all can benefit from doc components used in the MDX. 

**What is the chosen solution to this problem?**
Create a shared library of doc components, used as a dev-dependency by projects that require it. 

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/storybook-docs/custom.d.ts [hunk 1]
```diff
@@ -0,0 +1,327 @@
+declare module '*.png';
+
+declare module '*.svg' {
+	const content: React.FunctionComponent<React.SVGAttributes<SVGElement>>;
+	export default content;
+}
+
+declare module '@talend/icons' {
```
Comment (Benrajalu) [inline packages/storybook-docs/custom.d.ts:8]: Probably don't need this anymore

File: packages/storybook-docs/package.json [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    "@types/react-is": "^16.7.2",
+    "@types/styled-components": "^5.1.25",
+    "algoliasearch": "^4.13.0",
+    "babel-plugin-styled-components": "^1.13.3",
+    "browser-sync": "^2.27.9",
+    "browser-sync-webpack-plugin": "^2.3.0",
+    "chromatic": "^6.5.4",
+    "core-js": "^3.22.2",
+    "cypress": "^9.6.0",
+    "figma-js": "^1.15.0",
```
Comment (Benrajalu) [inline packages/storybook-docs/package.json:79]: Probably should be removed

File: packages/storybook-docs/package.json [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  "publishConfig": {
+    "access": "public"
+  },
+  "dependencies": {
+    "@talend/assets-api": "^1.2.0",
+    "@talend/design-system": "^2.7.0",
+    "@talend/design-tokens": "^2.4.0",
+    "classnames": "^2.3.1",
+    "modern-css-reset": "^1.4.0",
+    "reakit": "^1.3.11",
```
Comment (Benrajalu) [inline packages/storybook-docs/package.json:41]: Assumption, thus far

File: packages/storybook-docs/package.json [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    "@svgr/webpack": "^5.5.0",
+    "@talend/bootstrap-theme": "^6.39.0",
+    "@talend/icons": "^6.42.0",
+    "@talend/scripts-core": "^11.7.1",
+    "@talend/scripts-preset-react-lib": "^11.0.3",
+    "@types/classnames": "^2.3.1",
+    "@types/react": "^17.0.2",
+    "@types/react-dom": "^17.0.2",
+    "@types/react-is": "^16.7.2",
+    "@types/styled-components": "^5.1.25",
```
Comment (Benrajalu) [inline packages/storybook-docs/package.json:71]: Must be remooooved

File: packages/storybook-docs/package.json [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    "algoliasearch": "^4.13.0",
+    "babel-plugin-styled-components": "^1.13.3",
+    "browser-sync": "^2.27.9",
+    "browser-sync-webpack-plugin": "^2.3.0",
```
Comment (Benrajalu) [inline packages/storybook-docs/package.json:75]: Not needed

File: yarn.lock [hunk 6]
```diff
@@ -4184,7 +4184,7 @@
   dependencies:
     "@types/react" "^17"
 
-"@types/react@*", "@types/react@>=16.9.11", "@types/react@^16", "@types/react@^17", "@types/react@^17.0.2":
+"@types/react@*", "@types/react@>=16.9.11", "@types/react@^17", "@types/react@^17.0.2":
```
Comment (Benrajalu) [inline yarn.lock]: 🤔 maybe a `yarn deduplicate` or something I need to do. Weird that the lock changed when I added nothing new

File: packages/storybook-docs/.prettierrc [hunk 7]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (jsomsanith-tlnd) [inline packages/storybook-docs/.prettierrc:1]: Is this prettierrc necessary ? Doesn't it use the one in repo root ?

**Review by jsomsanith-tlnd — APPROVED** [review]

**Review by jmfrancois — APPROVED** [review]
could you please move the package.json to 1.0.0 ?

**Pull request merged**